awsUtilities/src/test/java/org/opensearch/migrations/aws/SigV4SigningTransformationTest.java

```diff
@@ -0,0 +1,131 @@
+package org.opensearch.migrations.aws;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.util.List;
+import java.util.stream.Stream;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import org.opensearch.migrations.transform.HttpJsonMessageWithFaultingPayload;
+import org.opensearch.migrations.transform.ListKeyAdaptingCaseInsensitiveHeadersMap;
+import org.opensearch.migrations.transform.PayloadAccessFaultingMap;
+import org.opensearch.migrations.transform.StrictCaseInsensitiveHttpHeadersMap;
+
+import lombok.SneakyThrows;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+
+import static org.opensearch.migrations.transform.JsonKeysForHttpMessage.INLINED_JSON_BODY_DOCUMENT_KEY;
+import static org.junit.jupiter.api.Assertions.*;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class SigV4SigningTransformationTest {
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    private static class MockCredentialsProvider implements AwsCredentialsProvider {
+        @Override
+        public AwsCredentials resolveCredentials() {
+            return AwsBasicCredentials.create("AKIAIOSFODNN7EXAMPLE", "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY");
+        }
+    }
+
+    private static Stream<Arguments> testCases() {
+        return Stream.of(
+            Arguments.of("GET", "/", null),
+            Arguments.of("PUT", "/index/_doc/1", "{\"field\":\"value\"}"),
+            Arguments.of("DELETE", "/index/_doc/1", null),
+            Arguments.of("HEAD", "/", null)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("testCases")
+    @SuppressWarnings("unchecked")
+    public void testSignatureProperlyApplied(String method, String path, String payload) {
+        // Setup
+        var mockCredentialsProvider = new MockCredentialsProvider();
+        var signer = new SigV4Signer(
+            mockCredentialsProvider,
+            "es",
+            "us-east-1",
+            "https",
+            () -> Clock.fixed(Instant.EPOCH, ZoneOffset.UTC)
+        );
+
+        // Create a mock HTTP message
+        var httpMessage = new HttpJsonMessageWithFaultingPayload();
+        httpMessage.setMethod(method);
+        httpMessage.setPath(path);
+
+        StrictCaseInsensitiveHttpHeadersMap strictHeaders = new StrictCaseInsensitiveHttpHeadersMap();
+        ListKeyAdaptingCaseInsensitiveHeadersMap headers = new ListKeyAdaptingCaseInsensitiveHeadersMap(strictHeaders);
```

Comment:
This class is overkill.  Assume that not all users will be using ListKeyAdaptingCaseInsensitiveHeadersMap.  Just send a regular map and make sure that it works.  It's useful to do sig4v signer tests in the replayer with these sorts of classes, but if I looked at this test by itself, I wouldn't use the sigv4 signer because it looks too complicated.